### DISCUSSION TITLE
file-scoped namespace

### DISCUSSION THREAD

**Discussion by HakamFostok** [xunit/xunit#2482] [General]
Hi, first I want to thank you for this great library I really love it and used always, It changed everything in the code I write (for better of course).

Just a small question. Are you interested in PR that apply "file-scoped namespace" (C# 10 feature) to the projects that target .netstandard (the projects which accept this feature)

if yes, I can do that, it is my pleasure to reciprocate to Xunit.
if no, then no problem at all, have a nice day :)

Thank you again.

**Comment by mu88** [comment]
Wouldn't that require a lot of conditional compiles/preprocessor directives? Because AFAIK xUnit targets `netstandard2.0`, but file-scoped namespaces require `net6.0` or higher 🤔

**Comment by Ergamon** [comment]
To my knowledge it is not really a framework feature. Why should it? I mean it is just a convinience function for better readability.

You can use file-scoped namespaces with netstandard2.0, even with net48.

Of course you have to set LangVersion to 10 to make it work.

**Reply by mu88** [reply]
Ah I didn't know that, thanks!

**Comment by bradwilson** [comment]
This was just recently done on the `main` branch.

I would not be interested in such a sweeping change on the `v2` branch. I'm trying to keep changes there to a bare minimum (and the code from there will not be ported into `main`).
